prow/plugins/projectmanager/projectmanager.go

```diff
@@ -0,0 +1,208 @@
+	labelsFetched := false
+	var labels []github.Label
+	for orgRepoName, managedOrgRepo := range orgRepos {
+		for projectName, managedProject := range managedOrgRepo.Projects {
+			for _, managedColumn := range managedProject.Columns {
+				if managedColumn.Org != "" && managedColumn.Org != pe.Repo.Owner.Login {
+					continue
+				}
+				if managedColumn.State != "" && managedColumn.State != pe.PullRequest.State {
+					continue
+				}
+				if len(managedColumn.Labels) != 0 {
+					if !labelsFetched {
+						// If labels are not yet fetched then get them as they are now required
+						// GetIssueLabels works for PRs as they are considered issues in the API
+						var err error
+						labels, err = gc.GetIssueLabels(pe.Repo.Owner.Login, pe.Repo.Name, pe.Number)
+						if err != nil {
+							return nil, err
+						}
+						labelsFetched = true
+					}
+					if !github.HasLabels(managedColumn.Labels, labels) {
+						continue
+					}
+				}
+				columnID := managedColumn.ID
+				// Currently this assumes columnID having a value if 0 means it is unset
+				// While it's highly unlikely that an actual project would have an ID of 0, given that
+				// these IDs are global across GitHub, this doesn't seem like an ideal solution.
+				if columnID == nil {
+					var err error
+					columnID, err = getColumnID(orgRepoName, projectName, managedColumn.Name, gc)
+					if err != nil {
+						return nil, err
+					}
+				}
+				matchedColumnIDs = append(matchedColumnIDs, *columnID)
+			}
+		}
+	}
+	return matchedColumnIDs, nil
+}
+
+func getColumnID(orgRepoName, projectName, columnName string, gc githubClient) (*int, error) {
+	var projects []github.Project
+	var err error
+	orgRepoParts := strings.Split(orgRepoName, "/")
+	switch len(orgRepoParts) {
+	case 2:
+		projects, err = gc.GetRepoProjects(orgRepoParts[0], orgRepoParts[1])
+	case 1:
+		projects, err = gc.GetOrgProjects(orgRepoParts[0])
+	default:
+		return nil, fmt.Errorf("could not determine org or org/repo from %s", orgRepoName)
+	}
+	if err != nil {
+		return nil, err
+	}
+	for _, project := range projects {
+		if project.Name == projectName {
+			columns, err := gc.GetProjectColumns(project.ID)
+			if err != nil {
+				return nil, err
+			}
+			for _, column := range columns {
+				if column.Name == columnName {
+					return &column.ID, nil
+				}
+			}
+			return nil, fmt.Errorf("could not find column %s in project %s", columnName, projectName)
+		}
+	}
+	return nil, fmt.Errorf("could not find project %s in org/repo %s", projectName, orgRepoName)
+}
+
+func addPRToColumn(gc githubClient, columnID int, pe github.PullRequestEvent) error {
+	// Create project card and add this PR
+	projectCard := github.ProjectCard{
+		ContentType: "PullRequest",
+		ContentID:   pe.PullRequest.ID,
+	}
+	_, err := gc.CreateProjectCard(columnID, projectCard)
+	if err != nil {
+		return err
```

Comment:
This will return errors when we add a duplicate card, right? If that is expected behavior we need to distinguish the duplicate card error and avoid returning an error in that case.

Reply:
Done